File: src/main/java/com/neovision/auth/service/AuthService.java
package com.neovision.auth.service;

import com.neovision.auth.dto.AuthResponse;
import com.neovision.auth.dto.LoginRequest;
import com.neovision.auth.dto.MicrosoftLoginRequest;
import com.neovision.auth.dto.RegistroRequest;

public interface AuthService {

    AuthResponse login(LoginRequest request);

    AuthResponse registro(RegistroRequest request);

    AuthResponse loginMicrosoft(MicrosoftLoginRequest request);
}

File: neovision-frontend/src/api/authService.ts
import api from './axios';

export interface LoginRequest {
  correo: string;
  contrasena: string;
}

export interface RegistroRequest {
  nombre: string;
  apellido: string;
  correo: string;
  contrasena: string;
  tipoDocumento: string;
  numeroDocumento: string;
  fechaNacimiento: string;
  telefono: string;
}

export interface MicrosoftLoginRequest {
  accessToken: string;
}

export interface AuthResponse {
  token: string;
  rol: string;
  correo: string;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

export const authService = {
  login: (data: LoginRequest) =>
    api.post<ApiResponse<AuthResponse>>('/auth/login', data),

  registro: (data: RegistroRequest) =>
    api.post<ApiResponse<AuthResponse>>('/auth/registro', data),

  loginMicrosoft: (data: MicrosoftLoginRequest) =>
    api.post<ApiResponse<AuthResponse>>('/auth/microsoft', data),
};
